HASH/mapas1.c

```diff
@@ -0,0 +1,437 @@
+				{
+					while(aux->sig != NULL)
+					{
+						aux = aux->sig;
+						nodo = malloc(sizeof(MAPA));
+						aux2->sig = nodo;
+						aux2 = nodo;
+						strcpy(nodo->LLAVE,aux->LLAVE);  //COPIAR LLAVE DE ELEMENTO
+						nodo->sig = NULL;
+					}
+				}
+			}
+
+		}
+	}
+	return;
+}
+
+/**
+ *================================================================================
+ *@brief     Esta funcion ingresa un nuevo elemento dentro del mapa especificado,
+ *           al ingresarlo, verifica que su llave sea unica contra las que estan
+ *           actualmente en el mapa y regresa el contenido del elemento ingresado.
+ *
+ *@entradas  MAPA *mapa[]:     Mapa donde se ingresa el elemento nuevo.
+ *           char contenido[]: Contenido del elemento nuevo.
+ *           char llave[]:     Llave del elemento nuevo.
+ *
+ *@salidas   char *res: Apuntador al contenido recien ingresado al mapa.
+ *
+ *@variables *nodo: Apuntador a estructura del mapa.
+ *           *aux:  Apuntador de soporte para "*nodo"
+ *           *aux2: Apuntador de soporte para "*nodo"
+ *           i:     Contador de FOR.
+ *           hash:  Valor de tabla hash del nuevo elemento.
+ *           valor: Valor ascii de la llave del nuevo elemento.
+ *           comp:  Resultado de comparacion para el oder del mapa.
+ *           *res:  Apuntador de resultado final de la funcion.
+ *
+ *@author    Luis Armando Vizcaino Magdaleno
+  *===============================================================================
+ */
+char *put(MAPA *mapa[], char contenido[], char llave[])
+{
+	MAPA *nodo, *aux, *aux2;
+	int i, hash, valor, comp;
+	char *res;
+	nodo = malloc(sizeof(MAPA));
+	strcpy(nodo->CONTENIDO,contenido);  //ASIGNACION DE CONTENIDO A NODO DE MAPA
+	strcpy(nodo->LLAVE,llave);  //ASIGNACION DE LLAVE A NODO DE MAPA
+	res = nodo->CONTENIDO; //VALOR DE RESPUESTA FINAL
+	verificar(nodo,mapa);  //VERIFICA QUE LA LLAVE SEA UNICA
+	valor = ascii(llave);
+	hash = valor%T;
+	aux = mapa[hash];
+	if(mapa[hash] == NULL)  //ASIGNACION DEL ELEMENTO DENTRO DE LA TABLA
+	{
+		mapa[hash] = nodo;
+		nodo->sig = NULL;
+		return (res);
+	}
+	else if(mapa[hash] != NULL)
+	{
+		comp = strcmp(aux->CONTENIDO,nodo->CONTENIDO);//ASIGNACION DEL ORDEN DE LOS ELEMENTOS
+		if(comp == 0)
+		{
+			mapa[hash] = nodo;
+			nodo->sig = aux;
+			return (res);
+		}
+		else if(comp < 0)
+		{
+			mapa[hash] = nodo;
+			nodo->sig = aux;
+			return(res);
+		}
+		if(comp > 0)
+		{
+			while(aux->sig != NULL)  //RECORRIDO Y ASIGNACION DEL ORDEN DE LOS ELEMENTOS ENCADENADOS
+			{
+				aux2 = aux;
+				aux = aux->sig;
+				comp = strcmp(aux->CONTENIDO,nodo->CONTENIDO);
+				if(comp == 0)
+				{
+					aux2->sig = nodo;
+					nodo->sig = aux;
+					return (res);
+				}
+				if(comp < 0)
+				{
+					aux2->sig = nodo;
+					nodo->sig = aux;
+					return (res);					
+				}
+				if(aux->sig == NULL)
+				{
+					aux->sig = nodo;
+					nodo->sig = NULL;
+					return (res);
+				}
+			}
+		}
+	}
+}
```

Comment:
Me marca un warning de compilación
gcc -c mapas1.c
mapas1.c:276:1: warning: control may reach end of non-void function [-Wreturn-type]
}